### PR TITLE
add lupa stubs

### PR DESCRIPTION
**Pull request by RedContritio**
- add stubs for [lupa](https://pypi.org/project/lupa/)

### CODE REVIEW

File: stubs/lupa/lupa/lua51.pyi [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    "as_itemgetter",
+    "as_attrgetter",
+    "lua_type",
+    "unpacks_lua_table",
+    "unpacks_lua_table_method",
+]
+
+LUA_MAXINTEGER: int
+LUA_MININTEGER: int
+LUA_VERSION: tuple[int, int]
```
Comment (srittau) [inline stubs/lupa/lupa/lua51.pyi:22]: These should probably be `Final`:

```suggestion
LUA_MAXINTEGER: Final[int]
LUA_MININTEGER: Final[int]
LUA_VERSION: Final[tuple[int, int]]
```

(Needs import from `typing`.)

File: stubs/lupa/lupa/lua51.pyi [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+class _LuaIter:
+    def __iter__(self) -> Iterator[object]: ...
+
+class _LuaTable:
+    def keys(self) -> _LuaIter: ...
+    def values(self) -> _LuaIter: ...
+    def items(self) -> _LuaIter: ...
+
+class _LuaNoGC: ...
+class _LuaObject: ...
```
Comment (srittau) [inline stubs/lupa/lupa/lua51.pyi:46]: I these are not available at runtime, they should be annotated with `@type_check_only` (from `typing`). Same below.

File: stubs/lupa/lupa/version.pyi [hunk 3]
```diff
@@ -0,0 +1 @@
+__version__: str
```
Comment (srittau) [inline stubs/lupa/lupa/version.pyi:1]:
```suggestion
__version__: Final[str]
```

**Review by srittau — CHANGES_REQUESTED** [review]
Thanks! A few remarks below. Also, whenever using `Any`, it should be documented why `Any` is necessary and what the possible types are.

**Comment by RedContritio** [comment]
> Thanks! A few remarks below. Also, whenever using `Any`, it should be documented why `Any` is necessary and what the possible types are.

I have removed almost all `Any`s, except for `Callable[..., Any]` used in decorators to represent arbitrary functions and `*args: Any, **kwargs: Any` used in variable arguments.

**Review by srittau — APPROVED** [review]

**Pull request merged**